prow/config/config.go

```diff
@@ -615,7 +633,16 @@ func (cfg *SlackReporter) DefaultAndValidate() error {
 		cfg.ReportTemplate = `Job {{.Spec.Job}} of type {{.Spec.Type}} ended with state {{.Status.State}}. <{{.Status.URL}}|View logs>`
 	}
 
-	if cfg.Channel == "" {
+	if cfg.ChannelConfig == nil {
+		cfg.ChannelConfig = map[string]string{}
+	}
+
+	if cfg.Channel != "" && cfg.ChannelConfig["*"] == "" {
+		cfg.ChannelConfig["*"] = cfg.Channel
```

Comment:
Could you also
* make this log a deprecation warning when `cfg.Channel` is set
* Error out when both `cfg.Channel` and `cfg.ChannelConfig["*"] ` are set?

Reply:
> Error out when both cfg.Channel and cfg.ChannelConfig["*"] are set?

I was going with a fallback-type approach but an error here is explicit and makes more sense.

Reply:
Cleaned it up a bit more.